### ISSUE TITLE
[BUG] DOS game Starfire v1.4 with Tseng ET4000 ISA video card 

### ISSUE THREAD

**Issue by Fenix770** [86Box/86Box#347]
Hello

I just tested the DOS game Starfire v1.4 both shareware and registered versions and  When I use the ET4000 ISA I get a black screen after selecting the sound card.
The game works without any issue with the Trident 8900 ISA card.

My configuration is: MR 386 DX (I also tested  386 SX machines).
4 mb RAM, DOS 5.0.  Here is the shareware version 1.4 of the game.

[Starfire v14 sharware.zip](https://github.com/86Box/86Box/files/2592275/Starfire.v14.sharware.zip)

**Comment by TC1995** [comment]
Have you tried TLIVESA? It is required for any VESA DOS app/game using ET4000AX.

**Comment by Fenix770** [comment]
Yes, but the problem is not related to VESA video modes. It uses the 320x200 256 colors standard VGA video mode. 
The game even works on OTI-037C, and this old card, also, has no VBE, and the only SVGA mode suported is 800x600 16 colors with the 37VESA.COM driver.

**Comment by TC1995** [comment]
Then I don't know to be honest.

**Comment by OBattler** [comment]
The game could be relying on the 256k video memory wraparound that ET4000AX does not do. That, or some CRTC write is not going through.

**Comment by OBattler** [comment]
It could also be the banking segment stuff. I've seen that cause problems in the past on the emulator.

**Comment by Fenix770** [comment]
Hello!

I write here again for this same game with a very different problem.  Now the game works but produces a "flashing" or "intermitent" image on the lower part of the screen. I tested the ET4000AX ISA and W32p PCI. both have the same problem.

Here is a video capture to see the screen bug.
[86Box v3.0 [feb61d7e, build 3023] - 101% - [i430LX] ASUS P_I-P5MP3 - Pentium_60 - Click to capture mouse 2021-06-18 21-31-37.zip](https://github.com/86Box/86Box/files/6679735/86Box.v3.0.feb61d7e.build.3023.-.101.-.i430LX.ASUS.P_I-P5MP3.-.Pentium_60.-.Click.to.capture.mouse.2021-06-18.21-31-37.zip)

Here is the hard disk image if needed:

[518mb HD P60 ET4000 w32p.zip](https://github.com/86Box/86Box/files/6679739/518mb.HD.P60.ET4000.w32p.zip)

**Comment by saipan7** [comment]
can confirm the game still and also produces a "flashing" or "intermitent" image on the lower part of the screen on below card:

- [ISA] ATI Mach8
- [ISA] ATI Mach32
- [ISA] ATI Korean VGA
- [ISA] ATI-18800-1
- [ISA] ATI-28800-5 (ATI VGA Charger)
- [ISA] ATI-28800-5 (ATI VGA Wonder XL)
- [ISA] Kasan Hangulmadang-16 VGA (Tseng Labs ET4000AX Korean)
- [ISA] Paradise PVGA1A
- [ISA] Paradise WD90C11-LR
- [ISA] Paradise WD90C30-LR
- [ISA] S3 ALL
- [ISA] Trigem Korean VGA (Tseng Labs ET4000AX Korean]
- [ISA] Tseng Labs ALL

- [VLB] ATI Mach32
- [VLB] Tseng Labs ALL
- [VLB] S3 ALL

![image](https://github.com/86Box/86Box/assets/127716019/47a19a08-07e3-4ac6-9e4e-81bea8cc0e71)


the game work fine on below card:

- [ISA] ATI Mach64GX
- [ISA] ATI-18800-5
- [ISA] Cirrus Logic ALL
- [ISA] G2 GC205
- [ISA] OAK ALL
- [ISA] Trident ALL
- [ISA] VGA
- [ISA] Video 7 VGA 1024i (HT208)

- [ISA16] INMOS XGA (XGA type: XGA-1 and XGA-2)
- [ISA16] Radius SVGA Multiview (HT209)
- [ISA16] Realtek RTG3106

- [VLB] ATI Mach64GX
- [VLB] Headland HT216-32
- [VLB] Trident TGUI 9400CXi
- [VLB] Trident TGUI 9440AGi

**Comment by OBattler** [comment]
Does this still occur?

**Comment by Fenix770** [comment]
I can confirm this still happen on both, the  ET-3000-AX and ET-4000AX ISA video card.


configuration:

dpi_scale = 0
vid_resize = 1
video_gl_shader =
vid_renderer = qt_software
video_gl_framerate = 50
video_gl_vsync = 1
rctrl_is_lalt = 1
enable_overscan = 1

[Machine]
machine = awardsx
cpu_family = cx486slc
cpu_speed = 20000000
cpu_multi = 1
cpu_use_dynarec = 0
time_sync = local
fpu_type = 387
fpu_softfloat = 0
cpu_manufacturer = 2
mem_size = 4096

[Video]
gfxcard = et4000ax

[Input devices]
mouse_type = msserial

[Sound]
sndcard = sbprov1
sound_type = int16
fm_driver = nuked

[Storage controllers]
hdc = ide_isa

Sound Blaster Pro v1]
base = 0220
irq = 7
dma = 1
opl = 1
receive_input = 0

[Tseng Labs ET4000AX (ISA)]
memory = 1024

**Comment by iamgreaser** [comment]
This thread mentions a few cards where this behaviour is "normal" due to the 256 KB wraparound not being implemented properly: https://www.vogons.org/viewtopic.php?f=63&t=96169

The S3 chips are probably supposed to wrap around, or at least the Trio64V2/DX should.

On what other cards is this behaviour normal?

**Comment by iamgreaser** [comment]
S3 Trio, Vision864, Vision964 should be fine since #3902 was merged (I tested a few in that range). It seems that the ViRGE (incl. Trio3D) is handling the wrapping just fine as well. Older stuff like the 86c928 and 86c911 is still not wrapping correctly, but that's ultimately going to require testing on a real card, although if there's a reference manual which has a way to enable and disable 256K wrapping (if such a feature exists in those cards) then that would also be helpful.
